code/obj/machinery/manufacturer.dm

```diff
@@ -1214,6 +1226,213 @@
+
+						if ("resume")
+							help["description"] = "Resumes building the current item."
+
+						if ("pause")
+							help["description"] = "Pauses bulding the current item."
+
+						if ("repeat")
+							help["description"] = "Sets whether or not the manufacturer is repeating building the current item based on the included state."
+							help["args"] = "data"
+
+				post_signal(help)
+				return
+
+			if ("status")
+				post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "status", "data" = "mode=[src.mode]&speed=[src.speed]&timeleft=[src.time_left]&repeat=[src.repeat]"))
+				return
+
+			if ("queue")
+				var/return_queue
+				for (var/datum/manufacture/I in src.queue)
+					return_queue += I.name + ","
+
+				post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "queue", "data" = return_queue))
+				return
+
+			if ("add")
+				var/item_name = signal.data["data"]
+				if (!item_name)
+					post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ERR#BADITEMNAME"))
+					return
+
+				var/datum/manufacture/item_bp
+				for (var/datum/manufacture/bp in src.available + src.download + src.drive_recipes + (src.hacked ? src.hidden : null))
+					if (bp.name == item_name)
+						item_bp = bp
+						break
+
+				if (!item_bp)
+					post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ERR#NOITEMBLUEPRINT"))
+					return
+
+				if (free_resource_amt > 0) // We do this here instead of on New() as a tiny optimization to keep some overhead off of map load - Also required for packets
+					claim_free_resources()
+
+				if (!check_enough_materials(item_bp))
+					post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ERR#NOMATERIALS"))
+					return
+
+				else if (src.queue.len >= MAX_QUEUE_LENGTH)
+					post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ERR#QUEUEFULL"))
+					return
+
+				else
+					src.queue += item_bp
+					src.begin_work(1)
+					post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ACK#APPENDED"))
+
+			if ("clear")
+				var/Qcounter = 1
+				for (var/datum/manufacture/M in src.queue)
+					if (Qcounter == 1 && src.mode == "working") continue
+					src.queue -= src.queue[Qcounter]
+
+				if (src.mode == "halt")
+					src.manual_stop = 0
+					src.error = null
+					src.mode = "ready"
+					src.build_icon()
+
+				post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ACK#CLEARED"))
+
+			if ("remove")
+				var/operation = text2num_safe(signal.data["data"])
+				if (!isnum(operation) || src.queue.len < 1 || operation > src.queue.len)
+					post_signal(list("address_1" = sender, "sender" = src.net_id, "command" = "term_message", "data" = "ERR#BADOPERATION"))
+					return
+
+				if(world.time < last_queue_op + 5)
```

Comment:
Consider using the `TIME` define here, to prevent issues with lag slowing things down.